src/gmp.c

```diff
@@ -10577,24 +10599,25 @@ results_xml_append_nvt (iterator_t *results, GString *buffer)
  * @param[in]  delta_results          Iterator for delta result to include, or
  *                                    NULL.
  * @param[in]  changed                Whether the result is a "changed" delta.
+ * @param[in]  cert_loaded            Whether the CERT db is loaded.
  */
 void
-buffer_results_xml (GString *buffer, iterator_t *results, task_t task,
-                    int include_notes, int include_notes_details,
-                    int include_overrides, int include_overrides_details,
-                    int include_tags, int include_tags_details,
-                    int include_details,
-                    const char *delta_state, iterator_t *delta_results,
-                    int changed)
+buffer_results_xml_cert (GString *buffer, iterator_t *results, task_t task,
```

Comment:
I'm not sure if passing the status of the CERT database around is the best idea here.
If avoiding the database queries helps, have you considered storing the state in a variable returned by `manage_cert_loaded ()` and having an update function called at the start of every GMP command 
(or every command that uses the CERT database)?

Maybe the scheduler could also update this regularly in the main process in case it becomes relevant for
scheduled tasks.

Reply:
Avoiding the database queries helps immensely when they are occurring for every result in a big report.

By having the option to pass the CERT status around we have the ability to avoid these queries when we can and need to, but also the ability to let it happen automatically the rest of the time.

I get that using some sort of variable in manage_cert_loaded is an option, but I don't really see that improving the situation -- we then need to deal with how to recheck the variable, which adds complexity.  Even if it was a better idea, I wouldn't think it's something we should be taking on in this PR.

Reply:
Okay, let's leave it the way it is now.